TranslucentTB/main.cpp

```diff
@@ -153,67 +169,185 @@ void PrintHelp()
 	}
 }
 
-void ParseOptions()
+void ParseSingleConfigOption(std::wstring arg, std::wstring value)
 {
-	// Set default values
-	opt.taskbar_appearance = ACCENT_ENABLE_BLURBEHIND;
-	opt.color = 0x00000000;
+	if (arg == L"accent")
+	{
+		if (value == L"blur")
+			opt.taskbar_appearance = ACCENT_ENABLE_BLURBEHIND;
+		else if (value == L"opaque")
+			opt.taskbar_appearance = ACCENT_ENABLE_GRADIENT;
+		else if (value == L"transparent" ||
+				 value == L"translucent")
+			opt.taskbar_appearance = ACCENT_ENABLE_TRANSPARENTGRADIENT;
+	}
+	else if (arg == L"color" ||
+			 arg == L"tint")
+	{
+		if (value.find(L'#') == 0)
+			value = value.substr(1, value.length() - 1);
+
+		size_t numchars = 0;
+		// heh make sure we don't run into overflow errors 
+		// TODO use proper range here and check for overflows. Write to logfile and warn user of error.
+		unsigned long long parsed = std::stoull(value, &numchars, 16);
+
+		opt.color =
+			(parsed & 0xFF000000) +
+			((parsed & 0x00FF0000) >> 16) +
+			(parsed & 0x0000FF00) +
+			((parsed & 0x000000FF) << 16);
+	}
+	else if (arg == L"opacity")
+	{
+		// TODO Same here. Should check range and warn user if the value doesn't fit.
+		size_t numchars = 0;
+		int parsed = std::stoi(value, &numchars, 10);
 
-	// Loop through command line arguments
-	LPWSTR *szArglist;
-	int nArgs;
+		if (parsed < 0)
+			parsed = 0;
+		else if (parsed > 255)
+			parsed = 255;
 
-	szArglist = CommandLineToArgvW(GetCommandLineW(), &nArgs);
+		forcedtransparency = parsed;
+	}
+}
 
-	for (int i = 0; i < nArgs; i++)
+void ParseConfigFile()
+{
+	std::wifstream configstream(configfile);
+
+	for (std::wstring line; std::getline(configstream, line); )
+	{
+		// Skip comments
+		size_t comment_index = line.find(L';');
+		if (comment_index == 0)
+			continue;
+		else
+			line = line.substr(0, comment_index);
+
+		size_t split_index = line.find(L'=');
+		std::wstring key = line.substr(0, split_index);
+		std::wstring val = line.substr(split_index + 1, line.length() - split_index - 1);
+
+		ParseSingleConfigOption(key, val);
+	}
+
+	if (forcedtransparency >= 0)
+	{
+		opt.color = (forcedtransparency << 24) +
+			(opt.color & 0x00FFFFFF);
+	}
+}
+
+void SaveConfigFile()
+{
+	using namespace std;
+	if (!configfile.empty())
```

Comment:
I think we should only save config is it is passed via command line (I will add a tray option too). Let the user tell it when to overwrite!